### DISCUSSION TITLE
Lesson 6:Cast Call for interacting with deployed contract

### DISCUSSION THREAD

**Discussion by miladad8** [Cyfrin/foundry-full-course-cu#1105] [General]
Hi.I'm trying to interact with a deployed contract using terminal in VS code. I can successfully deploy a contract. `cast send` command also works fine with `store()` function. but when I try to use `cast call` for `retrieve()` function, I get `0x` everytime.below is my code:

`cast call 0x90193C961A926261B756D1E5bb255e67ff9498A1 "retrieve()"`

but all I get is:

`0x`

I have also tried to store different values and then retrieve them, but nothing has changed.
I also check the `retrieve()` spelling and copy the exact function name from the main contract.

Any help would be appreciated.

**Comment by ArnaudBand** [comment]
Hello developers, I hope you are doing well. After deploying it successfully, I'm also trying to interact with the smart contract. When I use 
`cast send 0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266 "store(uint256)" 2011 --rpc-url $RPC_URL --private-key $PRIVATE_KEY`.
The data went to the blockchain the problem comes when I read from the blockchain.

`cast call 0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266 "retrieve()"`

it displays

`0x`
and when I run 

`cast --to-base 0x dec`

Got this error
`Error: 
could not parse hexadecimal value: cannot parse integer from empty string
`
